src/GraphQL/Types/Directives/LengthDirective.cs

```diff
@@ -0,0 +1,50 @@
+using System;
+
+namespace GraphQL.Types
+{
+    /// <summary>
+    /// Used to specify the minimum and/or maximum length for an input field or argument.
+    /// </summary>
+    public class LengthDirective : DirectiveGraphType
```

Comment:
Once we figure out all the details of the implementation, such as not calling `GetVariableValues` more than once for a document, then we could consider including this directive.  I'm not sure if it should be included directly in the main namespace or even in the same nuget package.  Perhaps it should be in a namespace called `GraphQL.Extensions.Directives` or something.  The thing is that it currently cannot be used without updating the validation rule set to include the proper rule.  So unlike `UrlGraphType` and similar extensions, additional configuration is necessary for this directive to function.  Thoughts?

Reply:
https://github.com/graphql-dotnet/graphql-dotnet/pull/2276#discussion_r574480038

> additional configuration is necessary for this directive to function

Not necessary, although most likely. The directive acts as metainformation. It can be used at execution stage, not just validation. Also this information is useful to the client himself. For example, in my case, I could completely do without the validation rule, because my clients are generated automatically by introspection. I get information about the `@length` directive and do checks in the clients themselves, that is, there is not even a network call and validation.

Also think about something like `@cached(expiration: Int)` directive. The server can declare through introspection directives that it does not make sense to query individual fields again (some global/static data). The server only marks the fields with a directive, but does not have any additional processing logic. The directive serves as a way to tell the client what to do and what not to do. This is food for thought for client.

Reply:
We should at least make it clear within the xml comments that the directive does not perform validation without adding the proper validation rule to the rule set.  Note that this rule will be required to run on cached queries also, so `ExecutionOptions.CachedDocumentValidationRules` needs to be set as well (if using caching).

Reply:
Done.